Automattic/woocommerce-payments
# Set min width for ECE Google Pay button in blocks checkout

Fixes #

#### Changes proposed in this Pull Request




This was discovered during our exploratory Express Checkout Element (ECE) testing. When testing with Storefront theme with the sidebar enabled, the checkout blocks page becomes quite narrow. This narrowness combined with WC Core's `grid-template-columns` on the `.wc-block-components-express-payment__event-buttons` container lead to the spacing for the Google Pay button to be too small and automatically hidden by Stripe.

To account for this we determined the minimum width the Google Pay button needs to render (208px) and replaced the calculated minimum width on the `grid-template-columns` with this fixed minimum width. 

**This applies only to narrow checkout blocks pages (Storefront with sidebar enabled) and when more than 2 express checkout options are enabled.**

**Before**
CSS: `grid-template-columns: repeat(auto-fit, minmax(calc(33% - 10px), 1fr));`
When three or more items were present in the Express Checkout (i.e. WooPay, Google Pay, and PayPal) the 33% - 10px was not wide enough for Google Pay to render leading to this display:
![Screenshot 2024-07-17 at 4 01 32 PM](https://github.com/user-attachments/assets/f817c956-ba8d-4298-8ec7-f83689fad490)
When breaking below the media query breakpoint the CSS becomes `grid-template-columns: 1fr` and renders as a single column and all buttons:
![Screenshot 2024-07-17 at 4 02 13 PM](https://github.com/user-attachments/assets/562e9141-8b5e-41ca-b371-22f9aa14c30f)

**After**
CSS: `grid-template-columns: repeat( auto-fit, minmax( 208px, 1fr ) ) !important;`
The 208px is determined through experimentation and was found to be the minimum size that still renders the Google Pay button on ECE. This now leads to a single column list rendering like this:
![Screenshot 2024-07-17 at 4 06 41 PM](https://github.com/user-attachments/assets/4e0cc82d-d543-4c75-8fd6-c0ba54991ffe)
When breaking below the media query breakpoint, there is a point where it goes into two columns before going back to a single column:
![Screenshot 2024-07-17 at 4 07 42 PM](https://github.com/user-attachments/assets/7002c558-ebc0-433b-b9f1-ed330137c205)

Here is what it looks like with the sidebar disabled which I think is a more common display:
![Screenshot 2024-07-17 at 4 09 57 PM](https://github.com/user-attachments/assets/16270fe7-a75f-4e37-9be0-ddd3f0132edc)

With sidebar and just two buttons:
![Screenshot 2024-07-17 at 4 11 20 PM](https://github.com/user-attachments/assets/7098f163-89a0-4b61-9f93-0f140bc5b62c)






#### Testing instructions





![Screenshot 2024-07-26 at 10 37 23 AM](https://github.com/user-attachments/assets/d563c089-8ccf-4c7f-9c03-6aa294330155)

* Have a store with the Storefront theme and a checkout blocks page with the sidebar enabled (see example screenshot above)
* Enable both WooPay and Google Pay / Apple Pay.
* In a browser where you will be presented with both options, add a product to cart and go to the checkout block page.
* Without this PR applied adjust the browser window width to get smaller until the Google Pay button disappears (like in the Before image above).
* Now with this PR applied repeat the same steps and confirm that the Google Pay button never disappears and instead bumps to a new row.

-------------------


**Post merge**
